### ISSUE TITLE
wrong root disk during the ingest bug

### ISSUE THREAD

**Issue by DK101010** [apache/cloudstack#4547]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
core,api
~~~

##### CLOUDSTACK VERSION

~~~
version > 4.14
~~~

##### OS / ENVIRONMENT

vmware env.

##### SUMMARY

VCenter api give a list of disks with a lexicographic order. CS expect that the root disk the first element of this list is. But this is not always the case.

##### STEPS TO REPRODUCE

~~~
create a vm with min. one data disk in vcenter, detach root disk, attach again, ingest it to cs
~~~

##### EXPECTED RESULTS

~~~
Check if first disk is really the root disk and if not search and use the root disk 
~~~

##### ACTUAL RESULTS

~~~
ingested vm can not start because wrong root disk
~~~

**Comment by DK101010** [comment]
PR https://github.com/apache/cloudstack/pull/4548

**Comment by shwstppr** [comment]
As discussed in #4548, there isn't any foolproof way to find root disk listUnmanagedInstance output or SDK APIs. PR #4591 has been created to add a new param in import API to allow the caller of API to assign a particular disk as ROOT volume.

**Comment by shwstppr** [comment]
Fixed with #4591
